src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -168,11 +168,10 @@ export async function mono_download_assets(): Promise<void> {
             countAndStartDownload(asset);
         }
 
-        // continue after the dotnet.runtime.js was loaded
-        await loaderHelpers.runtimeModuleLoaded.promise;
-
-        // continue after we know if memory snapshot is available or not
-        await runtimeHelpers.memorySnapshotSkippedOrDone.promise;
+        if (loaderHelpers.config.startupMemoryCache) {
+            // continue after we know if memory snapshot is available or not
+            await loaderHelpers.memorySnapshotSkippedOrDone.promise;
```

Comment:
There is other place where the promise is awaited, Do we want guard that one as well or skip the guard at all? https://github.com/dotnet/runtime/blob/fd6d5fd0a330a8a6fd286a3d924f4e832aaeb479/src/mono/wasm/runtime/loader/assets.ts#L214

Reply:
I tested some more and fixed and unified it